ladybug_display_schema/visualization.py

```diff
@@ -293,28 +286,75 @@ class GraphicContainer(NoExtraBaseModel):
-        if isinstance(geo, (Mesh2D, Mesh3D, Polyface3D)):
-            if isinstance(geo, (Mesh2D, Mesh3D)):
-                assert len(values) == len(geo.faces) or len(values) == \
-                    len(geo.vertices), 'Expected number of values ({}) to match ' \
-                    'number of faces ({}) or number of vertices ({}).'.format(
-                        len(values), len(geo.faces), len(geo.vertices))
-            else:  # it's a Polyface3D
-                assert len(values) == len(geo.face_indices) or len(values) == \
-                    len(geo.vertices), 'Expected number of values ({}) to match ' \
-                    'number of faces ({}) or number of vertices ({}).'.format(
-                        len(values), len(geo.face_indices), len(geo.vertices))
-        else:  # make sure that there are min and max points
-            min_point, max_point = obj_props.get('min_point'), obj_props.get('max_point')
-            assert min_point is not None and max_point is not None, \
-                'If min_point or max_point are unspecified, then geometry must be a ' \
-                'Mesh or Polyface to determine the bounding box.'
-            if isinstance(geo, (list, tuple)):
+        data_sets, geo = obj_props.get('data_sets'), obj_props.get('geometry')
+        for data in data_sets:
+            values = data.values
+            if isinstance(geo, (Mesh2D, Mesh3D, Polyface3D)):
+                if isinstance(geo, (Mesh2D, Mesh3D)):
+                    assert len(values) == len(geo.faces) or len(values) == \
+                        len(geo.vertices), 'Expected number of values ({}) to match ' \
+                        'number of faces ({}) or number of vertices ({}).'.format(
+                            len(values), len(geo.faces), len(geo.vertices))
+                else:  # it's a Polyface3D
+                    assert len(values) == len(geo.face_indices) or len(values) == \
+                        len(geo.vertices), 'Expected number of values ({}) to match ' \
+                        'number of faces ({}) or number of vertices ({}).'.format(
+                            len(values), len(geo.face_indices), len(geo.vertices))
+            else:
                 assert len(values) == len(geo), 'Expected one value per geometry ' \
                     'when geometry is a list. Number of values ({}) does not match ' \
                     'number of geometries ({}).'.format(len(values), len(geo))
         return obj_props
+
+
+class VisualizationSet(NoExtraBaseModel):
+    """A visualization set containing analysis and context geometry to be visualized."""
+
+    type: constr(regex='^VisualizationSet$') = 'VisualizationSet'
+
+    analysis_geometry: AnalysisGeometry = Field(
+        None,
+        description='An AnalysisGeometry object for spatial data that should be '
+        'displayed in the visualization.'
+    )
+
+    context_geometry: List[ALL_UNION] = Field(
```

Comment:
I see your point that it would make our lives easier as developers.

I was just thinking that, as a user, it might be really nice to have the flexibility to describe everything with ladybug-geometry objects. But, thinking about this more, this conversion from ladybug-geometry to ladybug-display is something that I can do automatically on the Python object. So I agree that we don't need to put this extra burden onto the core libraries here.